redis/src/tls.rs

```diff
@@ -0,0 +1,57 @@
+//! Defines types to use with the tls.
+
+use native_tls::{Certificate as NativeCertificate, Result};
+
+/// An X509 certificate.
+#[derive(Clone)]
+pub struct Certificate(NativeCertificate);
+
+impl Certificate {
+    /// Parses a DER-formatted X509 certificate.
+    pub fn from_der(der: &[u8]) -> Result<Certificate> {
+        let cert = NativeCertificate::from_der(der)?;
+        Ok(Certificate(cert))
+    }
+
+    /// Parses a PEM-formatted X509 certificate.
+    pub fn from_pem(pem: &[u8]) -> Result<Certificate> {
+        let cert = NativeCertificate::from_pem(pem)?;
+        Ok(Certificate(cert))
+    }
+
+    /// Returns the DER-encoded representation of this certificate.
+    pub fn to_der(&self) -> Result<Vec<u8>> {
+        let der = self.0.to_der()?;
+        Ok(der)
+    }
+}
+
+impl From<&Certificate> for NativeCertificate {
```

Comment:
I think we we should get rid of this `impl` and make it explicit at the caller sites.

Reply:
Done